ckanext/validation/plugin/__init__.py

```diff
@@ -261,12 +280,34 @@ def after_update(self, context, data_dict):
                 else:
                     # This is an actual package_update call, validate the
                     # resources if necessary
+                    #
+                    # (canada fork only):
+                    # NOTE: if a legit `package_update` or `package_patch` action happens,
+                    # not via one of the resource actions, we cannot prevent the resubmission of
+                    # all of the resources to Validation. `package_update` will not know which
+                    # resources have actually changed or not.
+                    #
+                    # With that said, the blueprints (package edit and create form) use `allow_partial_update`
+                    # context, so there will not actually be any resources when updating a package's metadata
+                    # in the web browser. This is only an issue with the API calls to `/api/action/package_update`
+                    # and `/api/action/package_patch`, and anywhere else in extensions that do not use the
+                    # `allow_partial_update` context.
+                    #
+                    # TODO: this will be solved with the upstream fix to IResourceController hooks
                     self._handle_validation_for_resource(context, resource)
 
         else:
             # This is a resource
             resource_id = data_dict[u'id']
 
+            # (canada fork only): delete the context key,value noting
+            # that the `resource_delete` action is happening.
+            # TODO: remove after upstream fix to IResourceController hooks
+            if '__resource_delete' in context:
+                del context['__resource_delete']
+            if '__resource_update' in context:
+                del context['__resource_update']
```

Comment:
same thing but shorter
```suggestion
            context.pop('__resource_delete', None)
            context.pop('__resource_update', None)
```